### PR TITLE
[Permissions] add foreign keys for subjects to sync jobs table

### PR DESCRIPTION
**Pull request by thenamankumar**
closes: https://github.com/sourcegraph/sourcegraph/issues/46911

This replaces repository_id and user_id integer columns to foreign keys to their corresponding tables. 

This allows for jobs to be automatically deleted upon the deletion of the user or repo. 

TODO in a separate PR, also integration with user, repo soft delete events.

## Test plan

- turn on db worker and wait for jobs to be scheduled
- delete a user/repo
- check for jobs to be removed.

### CODE REVIEW

File: internal/database/permission_sync_jobs_test.go [hunk 1]
```diff
@@ -337,3 +358,74 @@ func TestPermissionSyncJobs_CancelQueuedJob(t *testing.T) {
...
+	require.Len(t, jobs, 1)
+
+	// Deleting repo
+	_, err = db.ExecContext(context.Background(), fmt.Sprintf(`DELETE FROM repo WHERE id = %d`, int(repo1.ID)))
+	require.NoError(t, err)
+
+	// Checking that the job is deleted.
+	jobs, err = store.List(ctx, ListPermissionSyncJobOpts{RepoID: int(repo1.ID)})
+	require.NoError(t, err)
+	require.Len(t, jobs, 0)
```
Comment (sashaostrikov) [inline internal/database/permission_sync_jobs_test.go:395]: you may as well use:
```suggestion
	require.Empty(t, jobs)
```

File: internal/database/permission_sync_jobs_test.go [hunk 2]
```diff
@@ -337,3 +358,74 @@ func TestPermissionSyncJobs_CancelQueuedJob(t *testing.T) {
...
+	require.Len(t, jobs, 1)
+
+	// Deleting user
+	err = usersStore.HardDelete(ctx, user1.ID)
+	require.NoError(t, err)
+
+	// Checking that the job is deleted.
+	jobs, err = store.List(ctx, ListPermissionSyncJobOpts{UserID: int(user1.ID)})
+	require.NoError(t, err)
+	require.Len(t, jobs, 0)
```
Comment (sashaostrikov) [inline internal/database/permission_sync_jobs_test.go:430]: you may as well use:
```suggestion
	require.Empty(t, jobs)
```

**Review by sashaostrikov — APPROVED** [review]
LGTM!

Did you also do `sg migration up->down->up` to test them?

Added a couple of suggestions in tests.

Feel free to just batch apply my nit GrAmMaTiCaL comments in tests so you don't lose your time :)

**Review by mrnugget — COMMENTED** [review]
Good tests and comments!

**Pull request merged**